lib/resque/worker.rb

```diff
@@ -228,7 +228,7 @@ def work(interval = 5.0, &block)
           break if interval.zero?
           log_with_severity :debug, "Sleeping for #{interval} seconds"
           procline paused? ? "Paused" : "Waiting for #{queues.join(',')}"
-          sleep interval
+          @signal_handler.handle_signal(interval)
```

Comment:
applied a change like https://github.com/resque/resque/pull/1502/commits/4ac5b36b7eb3f6cb0a58c40f41af47e6ea25a4a4.

> can we do a non-blocking select so it doesn't sleep and just returns if there was no signal yet? and keep the sleep here where it was

During sleeping, we can not get signal, then. I felt `@signal_handler.handle_signal(interval)` here is still better, but I added `@signal_handler.handle_signal(0)` for the case `work_one_job` returns true.

Also, I added `handle_signal` in `perform_with_fork` for during waiting a forked child finishes.

Please note that we still have problems that we can not handle signal `fork_per_job?` is false. Hmm.